src/test/java/com/hotels/mutantswarm/exec/OutcomeTest.java

```diff
@@ -54,13 +63,54 @@ public void testStateKilled(){
   
   @Test
   public void testStateSurvived(){
-    List<TestOutcome> testOutcomes = asList(testOutcome1, testOutcome2, testOutcome3);
     when(testOutcome1.getState()).thenReturn(MutantState.SURVIVED);
     when(testOutcome2.getState()).thenReturn(MutantState.SURVIVED);
     when(testOutcome3.getState()).thenReturn(MutantState.SURVIVED);
     
     Outcome outcome = new Outcome(mutant, mutation, testOutcomes);
     assertThat(outcome.getState(), is(MutantState.SURVIVED));
   }
+  
+  @Test
+  public void checkToString(){
+    Outcome outcome = new Outcome(mutant, mutation, testOutcomes);
+    String result = outcome.toString();
+    assertThat(result, is("Outcome [mutant=mutant, mutation=mutation, state=SURVIVED, testOutcomes=[testOutcome1, testOutcome2, testOutcome3]]"));
+  }
+  
+  @Test
+  public void equalsNull() {
+    Outcome outcome = new Outcome(mutant, mutation, testOutcomes);
+    assertThat(outcome.equals(null), is(false));
+  }
+
+  @Test
+  public void equalsMutantNull() {
+    Outcome outcome = new Outcome(null, mutation, testOutcomes);
+    Outcome outcome2 = new Outcome(mutant, mutation, testOutcomes);
+    assertThat(outcome.equals(outcome2), is(false));
+  }
+  
+  @Test
+  public void equalsMutationNull() {
+    Outcome outcome = new Outcome(mutant, null, testOutcomes);
+    Outcome outcome2 = new Outcome(mutant, mutation, testOutcomes);
+    assertThat(outcome.equals(outcome2), is(false));
+  }
+  
+  @Test
+  public void equalSame() {
+    Outcome outcome = new Outcome(mutant, mutation, testOutcomes);
+    Outcome outcome2 = new Outcome(mutant, mutation, testOutcomes);
+    assertThat(outcome.equals(outcome2), is(true));
+    assertThat(outcome.hashCode() == outcome2.hashCode(), is(true));
```

Comment:
I see there's been some discussion on this but doing `assertThat(outcome.hashCode(), is(outcome2.hashCode())` is much better as you'll get a proper error message with the difference if this fails. Currently it will just say "wasn't true" which isn't very useful for debugging the problem.